wordpress-mobile/AztecEditor-iOS
# MediaAttachment: Encapsulating Draw Methods!

### Description:
In this PR we're:

- Updating the MediaAttachment's delegate signature (a bit more swifty!)
- Replacing few `if return` clauses with `guard`. Why? Because we reduce the indentation level, it's easier to read, and looks beautiful.
- The **MediaAttachment**'s draw method has been split into methods. (Note that this was partially complete prior to this PR!)
- **CGRect**'s methods (min, max) have been wired, and computation here and there has been simplified.
- No new functionality added. Just moving code arround!

### To test:
1. Verify that the unit tests pass okay
2. Add a Media Attachment:
2.1. Verify that the progress bar works properly
2.2. Verify that during the upload OP, there's an overlay
3. Press over the attachment
3.1. Verify that an overlay gets displayed
3.2. Verify that there's an asset rendered at the center of the image
3.3. Verify that below the center's asset, there's a text label!

I was in the neighborhood, and needed to change few things. This PR will be followed up by another one, in charge of rendering Figure / Figcaption elements.

cc @diegoreymendez @SergioEstevao 

Thanks in advance!
